### PR TITLE
Fix the user's wallet route

### PR DESCRIPTION
**Pull request by chinins**
## Description

Fix the user's wallet route & edit tokens dialog & form

**New stuff** ✨

- add `UserTokenEditDialogForm` 
- add tokens to `InMemoryCache` user
- add balance fetching in `getTokenData` for `token` resolver

**Changes** 🏗

- change balance  from resolver function to field in `Token`
- fix several small things in `token` part of `user` resolver 
- update `UserTokens` query & typedef

**Deletions** ⚰️

- delete `balance` function in `token` resolver

## TODO

- none

Resolves DEV-168

Screenshot 2021-02-02 at 15 41 52
Screenshot 2021-02-02 at 15 41 10

### CODE REVIEW

File: src/data/graphql/typeDefs.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ export default gql`
     symbol: String!
     iconHash: String
     verified: Boolean!
-    balance(walletAddress: String!): String!
+    balance: String
```
Comment (alicjakujawa) [inline src/data/graphql/typeDefs.ts:72]: Are we sure that balance CAN be a null? I feel that here we should have String!

File: src/data/graphql/typeDefs.ts [hunk 2]
```diff
@@ -124,7 +124,7 @@ export default gql`
 
   extend type User {
     reputation(colonyAddress: String!, domainId: Int): String!
-    tokens: [Token!]!
+    tokens(walletAddress: String): [Token!]!
```
Comment (alicjakujawa) [inline src/data/graphql/typeDefs.ts:127]: Can we query tokens without walletAddress?

File: src/modules/dashboard/components/Wallet/UserTokenEditDialogForm.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+import { UserTokens, OneToken, useLoggedInUser } from '~data/index';
+import { Address } from '~types/index';
+import { createAddress } from '~utils/web3';
+
+import styles from './UserTokenEditDialogForm.css';
+
+const MSG = defineMessages({
+  title: {
+    id: 'dashboard.UserTokenEditDialog.UserTokenEditDialogForm.title',
```
Comment (alicjakujawa) [inline src/modules/dashboard/components/Wallet/UserTokenEditDialogForm.tsx:24]: I think those ids are to change, UserTokenEditDialog is not needed here

Comment (chinins) [reply src/modules/dashboard/components/Wallet/UserTokenEditDialogForm.tsx:24]: Why? I thought that this was a convention that we are using, e.g.: 
`dashboard.TransferFundsDialog.TransferFundsDialogForm.title` (and also other dialog forms)
I am happy to change, just want to understand what would be the difference in this case.

File: src/data/graphql/typeDefs.ts (hunk 2 above)
Comment (chinins) [reply src/data/graphql/typeDefs.ts:127]: for the user no, you are right - good catch.

File: src/data/graphql/typeDefs.ts (hunk 1 above)
Comment (chinins) [reply src/data/graphql/typeDefs.ts:72]: yeah, probably

File: src/modules/dashboard/components/Wallet/UserTokenEditDialog.tsx [hunk 4]
```diff
@@ -50,7 +51,7 @@ const UserTokenEditDialog = ({ cancel, close, walletAddress }: Props) => {
      * the user and wire it up for launch
      */
     // @ts-ignore
-    <TokenEditDialog
+    <DialogForm
```
Comment (rdig) [inline src/modules/dashboard/components/Wallet/UserTokenEditDialog.tsx:54]: Haha. You just fixed it!

So you can safely remove the above comment :)

File: src/modules/dashboard/components/Wallet/UserTokenEditDialogForm.tsx (hunk 3 above)
Comment (rdig) [reply src/modules/dashboard/components/Wallet/UserTokenEditDialogForm.tsx:24]: Alicja is right here. this should be:
```
dashboard.Wallet.UserTokenEditDialogForm.title
```

Also the component should have a `displayName`, which should be `dashboard.Wallet.UserTokenEditDialogForm`

**Review by rdig — APPROVED** [review]
I'm good with this.

Pre-approving, but please take care of the stuff Alicja mentioned before merging!

**Review by ArmandoGraterol — APPROVED** [review]
It's looking dandy for me :fire: Great work!

**Pull request merged**